src/components/CreateCourse/CreateCourse.jsx

```diff
@@ -0,0 +1,148 @@
+import { useState } from 'react';
+import Form from 'react-bootstrap/Form';
+import Button from '../../common/Button/Button';
+import Input from '../../common/Input/Input';
+import styles from '../CreateCourse/CreateCourse.module.css';
+import AuthorItem from './components/AuthorItem/AuthorItem';
+import formatCreationDate from '../../helpers/formatCreationDate';
+import createGuid from '../../helpers/createGuid';
+
+const CreateCourse = ({ course, setCourses, author, setAuthor }) => {
+  let [inputs, setInputs] = useState({});
+  let [authorInput, setAuthorInput] = useState('');
+  let [formAuthors, setFormAuthors] = useState([]);
+
+  let authorId = createGuid();
+  let courseId = createGuid();
+
+  const addAuthorToForm = (authorName, event) => {
+    event.stopPropagation();
+    if (formAuthors.includes(authorName)) return;
+
+    setFormAuthors([...formAuthors, authorName]);
+  };
+
+  const deleteAuthor = (authorName) => {
+    setFormAuthors(formAuthors.filter((author) => author !== authorName));
+  };
+
+  const handleChange = (event) => {
+    const name = event.target.name;
+    const value = event.target.value;
+    setInputs((values) => ({ ...values, [name]: value }));
+  };
+
+  const handleAuthorChange = (event) => {
+    setAuthorInput(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const newCourses = [
+      ...course,
+      {
+        id: courseId,
+        creationDate: formatCreationDate(),
+        authors: formAuthors,
+        ...inputs,
+      },
+    ];
+    setCourses(newCourses);
+
+    setInputs({});
+  };
+
+  const createAuthor = () => {
+    const shouldReturn = (element) => {
+      return element.name === authorInput;
+    };
+
+    author.some(shouldReturn);
```

Comment:
1. Well, no sense to declare this function in different way, for methods that accept some callback you should also declare methods inside.
2. Should we leave this logic? I mean, you are not using `some` result anywhere